File: fake-store-quarkus/src/main/java/org/arielzarate/infrastructure/rest/dto/ProductDto.java
package org.arielzarate.infrastructure.rest.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDto {
    private Integer id;
    private String title;
    private Double price;
    private String description;
    private String category;
    private String image;
    private RatingDto rating;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RatingDto {
        private Double rate;
        private Integer count;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/fake_api_hexagonal/infraestructure/rest/model/ProductDto.kt
package com.arielZarate.fake_api_hexagonal.infraestructure.rest.model

data class ProductDto (
    val id: Int,
    val title: String,
    val price: Double,
    val description: String,
    val category: String,
    val image: String,
    val rating: RatingDto
)